WSLGd/FontMonitor.cpp

```diff
@@ -43,39 +43,71 @@ wslgd::FontFolder::~FontFolder()
     }
 }
 
-void wslgd::FontFolder::ExecuteShellCommand(const char *cmd)
+bool wslgd::FontFolder::ExecuteShellCommand(std::vector<const char*>&& argv)
 {
     bool success = false;
+    int childPid = -1, waitPid = -1;
+    std::string cmd;
+
+    try {
+        THROW_LAST_ERROR_IF((childPid = fork()) < 0);
+        if (childPid == 0) {
+            /* move this process to own process group to avoid interfere with Process Monitor */
+            THROW_LAST_ERROR_IF(setpgid(0, 0) < 0);
+            THROW_LAST_ERROR_IF(execvp(argv[0], const_cast<char *const *>(argv.data())) < 0);
+        } else if (childPid > 0) {
+            /* move child to own process group to avoid interfere with Process Monitor */
+            THROW_LAST_ERROR_IF(setpgid(childPid, childPid) < 0);

```

Comment:
yes this was suggested at some samples, to avoid race condition if parent reach to waitpid() before the child execute setgpid(), but I couldn't confirm that's really needed though, but it should hurt, so did it, just in case.